library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/MutablePack.java

```diff
@@ -293,6 +294,14 @@ default void putImage(Identifier id, Supplier<NativeImage> imageSupplier) {
 		}));
 	}
 
+	/**
+	 * Adds a pack overlay to the mutable pack.
+	 *
+	 * @param overlay the overlay name
+	 * @param pack    the pack
+	 */
+	void putOverlay(@NotNull String overlay, ResourcePack pack);
```

Comment:
@ix0rai @TheGlitch76 Im not too sure about this. Should it be a `ResourcePack`, where any pack can be part of the mutable pack, or does `Consumer<ResourcePack>` make more sense where the `MutablePack` implementation creates the pack in the way that it wants? Both options have their advantages.

Reply:
I have zero expertise in this area, it's really up to you...